solution/index.js

```diff
@@ -0,0 +1,423 @@
+                alert("Task must has info");
+                break;
+            }
+            task.innerText=text;
+            ul=document.getElementById("todo");
+            ul.insertBefore(task,ul.firstChild); 
+            document.getElementById("add-to-do-task").value="";
+            list=JSON.parse(localStorage.getItem("tasks"));//push the task to local storage
+            list["todo"].unshift(text);
+            localStorage.setItem("tasks",JSON.stringify(list));
+            break;
+        case "2":
+            text=document.getElementById("add-in-progress-task").value;
+            if(checkEmpty(text))
+            {
+                alert("Task must has info");
+                break;
+            }
+            task.innerText=text;
+            ul=document.getElementById("in-progress");
+            ul.insertBefore(task,ul.firstChild); 
+            document.getElementById("add-in-progress-task").value="";
+            list=JSON.parse(localStorage.getItem("tasks"));//push the task to local storage
+            list["in-progress"].unshift(text);
+            localStorage.setItem("tasks",JSON.stringify(list));
+            break;
+        case "3":
+            text=document.getElementById("add-done-task").value;
+            if(checkEmpty(text))
+            {
+                alert("Task must has info");
+                break;
+            }
+            task.innerText=text;
+            ul=document.getElementById("done");
+            ul.insertBefore(task,ul.firstChild); 
+            document.getElementById("add-done-task").value="";
+            list=JSON.parse(localStorage.getItem("tasks"));//push the task to local storage
+            list["done"].unshift(text);
+            localStorage.setItem("tasks",JSON.stringify(list));
+            break;
+    }
+}
+//function that check if task is empty
+function checkEmpty(text)
+{
+    if (!text.replace(/\s/g, '').length) {
+        return true;
+    }
+    return false;
+}
+//function that changing task
+function changeTask(event)
+{
+    let task =this;
+    listId=task.parentElement.id;
+    deleteTaskByText(task.innerText,listId) 
+    task.contentEditable=true;
+    task.addEventListener('blur',(event)=>{
+        task.contentEditable=false;
+        list=JSON.parse(localStorage.getItem("tasks"));//push the task to local storage
+        list[listId].unshift(task.innerText);
+        localStorage.setItem("tasks",JSON.stringify(list));
+    })
+}
+
+//function that delete task given by text
+function deleteTaskByText(text,listName)
+{
+    let check=true;
+    let tasksJson=JSON.parse(localStorage.getItem("tasks"));
+    let list=tasksJson[listName];
+    for (let i = 0; i < list.length; i++) {
+        if(text==tasksJson[listName][i]&&check)
+        {
+            tasksJson[listName].splice(i,1);
+            check=false;
+        }
+    }
+    localStorage.setItem("tasks",JSON.stringify(tasksJson));
+}
+
+
+
+//add and remove key press event
+let moveTaskLi;
```

Comment:
You don't need a global variable to know which task element is hovered. Find another way